program_admin/instructions.py

```diff
@@ -188,6 +189,42 @@ def delete_price(
     )
 
 
+def init_price(
+    funding_key: PublicKey,
+    program_key: PublicKey,
+    price_key: PublicKey,
+    exponent: int,
+    price_type: int = PRICE_TYPE_PRICE,
+) -> TransactionInstruction:
+    """
+    Pyth program init_price instruction
+
+    accounts:
+    - funding account (signer, writable)
+    - price account (signer, writable)
+    """
+    layout = Struct(
+        "version" / Int32ul, "command" / Int32sl, "exponent" / Int32sl, "type" / Int32ul
+    )
+    data = layout.build(
+        dict(
+            version=PROGRAM_VERSION,
+            command=COMMAND_INIT_PRICE,
+            exponent=exponent,
+            type=price_type,
+        )
+    )
+
+    return TransactionInstruction(
```

Comment:
You need to do it with the new interface if you want this to be compatible with the multisig:
```python
permissions_account = PublicKey.find_program_address([b"permissions"], program_id= program_key)[0]

 return TransactionInstruction(
     data=data,
     keys=[
         AccountMeta(pubkey=funding_key, is_signer=True, is_writable=True),
        AccountMeta(pubkey=price_key, is_signer=False, is_writable=True),
        AccountMeta(pubkey=permissions_account, is_signer=False, is_writable=False)]
        )
   ```
   Basically there's an extra account that contains the admin permission, and `price_key` no longer needs to sign